### PR TITLE
Adding event id & added removeEvent functionality

### PR DESCRIPTION
**Pull request by program-sam**
As suggested I added the removeEvent functionality. I did this by allowing for event identifiers, by which events can be searched, and removed again after they are added. I added an example and updated relevant parts of code where events were created, but I did not refactor the entire event design, as that would probably require some more discussion.

what do you think @Zemledelec ?

### CODE REVIEW

**Comment by program-sam** [comment]
I am not really sure what happened in src/og/input/KeyboardHandler.js, why it's all marked as 'changed'

File: src/og/renderer/RendererEvents.js [hunk 1]
```diff
@@ -264,29 +264,30 @@ class RendererEvents extends Events {
...
      * @param {number} [priority] - Event callback priority
      */
-    on(name, p0, p1, p2, p3) {
+    on(name, callback, keyCode, sender, priority, id) {
         if (name === "keypress" || name === "charkeypress" || name === "keyfree") {
-            this._keyboardHandler.addEvent(name, p2, p1, p0, p3);
+            this._keyboardHandler.addEvent(name, callback, keyCode, sender, priority, id);
         } else {
-            super.on(name, p0, p1, p2);
+            super.on(name, callback, keyCode, sender);
```
Comment (Zemledelec) [inline src/og/renderer/RendererEvents.js:277]: Third parameter is a sender in the parent class

Comment (Zemledelec) [reply src/og/renderer/RendererEvents.js:277]: ...this is why parameters called p1,p2 etc

**Review by Zemledelec — CHANGES_REQUESTED** [review]
1) I think that the better idea is to extract keyboard function for binding in a separate method. For instance onKeyPressed(), onKeyDown et, or` onKey(["pressed", "down", "up"], key code, callback, sender, priority)`.

2) No need ID. Event id is already stamped inside super event stamp methos, I think it is better idea to support off method offKey(callback), because callback already has an id inside `__ogstamp_`_ or something...so removeEvent = function (events, callback), could be invoke two parameters and depends on a callback, not id, because id is a stamp stored in the function of the callback

FYI @pavletto

File: src/og/renderer/RendererEvents.js (hunk 1 above)
Comment (Zemledelec) [reply src/og/renderer/RendererEvents.js:277]: @program-sam take a look above, I think that the separate keyboard event method convenient way to refactor this "facade"

**Comment by Zemledelec** [comment]
Just in case, for KeyboardHandler, it would be better not to change all code, because it's hard to review changings :) Just in case, it's a simple module, but for a better experience, I suggest somehow avoiding all code replacing, for example, adding new code in the end...

File: src/og/renderer/RendererEvents.js (hunk 1 above)
Comment (program-sam) [reply src/og/renderer/RendererEvents.js:277]: Ok I get the event ID part. I wasn't aware that these objects were stamped, but that's a really cool implementation! And yes this stamp does remove the need for a custom id, I'll remove those changes.on / off functions for all type of events work, but callbacks are not yet stamped for  "keypress", "charkeypress" and "keyfree" events If i Interpret things correctly? Which is why the off function for keyboard events was not yet implemented? Yes this stamp does remove the need for a custom id, I'll remove those changes.

But for the off function, that requires that you still have a reference to the original callback? The callback that was stamped. In a case like `events.on("keyfree", e => alert('space bar pressed!')` you will lose the reference to the callback function and won't be able to pass it again in the off function right?

I am not really sure what you mean with the idea "to extract keyboard function for binding in a separate method"

**Comment by Zemledelec** [comment]
I'll try to add what I think about in another branch. It's better to show it in code. 
FYI @program-sam

**Comment by Zemledelec** [comment]
Replaced to https://github.com/openglobus/openglobus/pull/753

**Pull request closed without merging**